WordPress/Classes/ViewRelated/Suggestions/SuggestionsTableView.swift

```diff
@@ -97,6 +131,54 @@ extension SuggestionType {
         }
     }
 
+    // MARK: - Private
+
+    private func searchResults(searchQuery: String, suggestions: [Any], suggestionType: SuggestionType) -> [Any] {
+        var searchResults: [Any]
+        if !searchQuery.isEmpty {
+            let predicate = self.predicate(for: searchQuery)
+            searchResults = NSMutableArray(array: suggestions).filtered(using: predicate)
+        } else {
+            searchResults = suggestions
+        }
+        if suggestionType == .mention {
+            searchResults = self.moveProminentSuggestionsToTop(searchResults: searchResults, prominentSuggestionsIds: prominentSuggestionsIds ?? [])
+        }
+        return searchResults
+    }
+
+    private func moveProminentSuggestionsToTop(searchResults: [Any], prominentSuggestionsIds ids: [NSNumber]) -> [Any] {
```

Comment:
I like this version with no side effects! 👏 

Is there any particular reason we're going with `searchResults: [Any]` and returning `[Any]` over enforcing the `UserSuggestion` type?

Would `private func moveProminentSuggestionsToTop(searchResults: [UserSuggestion], prominentSuggestionsIds ids: [NSNumber]) -> [UserSuggestion]` strengthen the function's purpose?

Reply:
>Is there any particular reason we're going with searchResults: [Any] and returning [Any] over enforcing the UserSuggestion type?
I think I just haven't paid attention to this. `searchResults` type from ObjC is `NSArray` ( aka `[Any]` ). But I agree it should be `[UserSuggestion]` for `moveProminentSuggestionsToTop`